### PR TITLE
Add support for the system User Class

### PR DESCRIPTION
**Pull request by halkeye**
Add a User store to specifically handle users (which don't need a unique store and doesn't extend from Item/ItemGroup)

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/uniqueid/implv2/UserIdStore.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import hudson.model.Job;
+import hudson.model.PersistenceRoot;
+import hudson.model.Run;
+import hudson.model.User;
+import org.jenkinsci.plugins.uniqueid.IdStore;
+
+/**
+ * Manages Unique IDs for User.
+ * We could make a unique file for every user. But a user already has a jenkins wide unique id
+ * that we can just tap into
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/uniqueid/implv2/UserIdStore.java]:
```suggestion
 * that we can just tap into
 * @since TODO
```

Comment (halkeye) [reply src/main/java/org/jenkinsci/plugins/uniqueid/implv2/UserIdStore.java]: Bad copy and paste

**Comment by halkeye** [comment]
@oleg I can create a ticket tomorrow if needed. I've moved to mobile now

**Comment by oleg-nenashev** [comment]
retriggering CI

**Comment by halkeye** [comment]
ah, doesn't fail with mvn test, but does with -Djenkins.version=2.164.1

**Comment by halkeye** [comment]
Found it, test needed user to be saved between reload

**Comment by halkeye** [comment]
@oleg-nenashev all fixed btw

**Review by jtnord — APPROVED** [review]

**Pull request merged**